src/Microsoft.Sbom.Parsers.Spdx22SbomParser/Generator.cs

```diff
@@ -185,15 +195,8 @@ public GenerationResult GenerateJsonDocument(SbomPackage packageInfo)
 
     private bool ShouldWeKeepTheExistingId(string spdxId)
     {
-        switch (configuration.ManifestToolAction)
-        {
-            case ManifestToolActions.Aggregate:
-                return true;
-            case ManifestToolActions.Generate:
-                return spdxId.Equals(Constants.RootPackageIdValue, StringComparison.OrdinalIgnoreCase);
-        }
-
-        return false;
+        return keepAllExistingIds ||
```

Comment:
Quick note: Prior to #1139, this code this method replaced _always_ checked for equality with `Constants.RootPackageIdValue`. After this PR, the pre- #1139 behavior changes only if `keepAllExistingIds` is true